File: services/subtraction-service/src/main/java/com/calculator/subtraction/domain/CalculationResult.java
package com.calculator.subtraction.domain;

import java.math.BigDecimal;
import java.time.Duration;
import java.time.Instant;

public record CalculationResult(
    BigDecimal result,
    Duration processingTime,
    boolean cached,
    Instant timestamp
) {
    public CalculationResult(BigDecimal result, Duration processingTime, boolean cached) {
        this(result, processingTime, cached, Instant.now());
    }
}

File: services/addition-service/src/main/kotlin/com/calculator/addition/domain/CalculationResult.kt
package com.calculator.addition.domain

import java.math.BigDecimal
import java.time.Duration
import java.time.Instant

data class CalculationResult(
    val result: BigDecimal,
    val processingTime: Duration,
    val cached: Boolean,
    val timestamp: Instant = Instant.now()
)